---
title: "Local Ancestry Chromosome Painting Plot"
output: html_notebook
---


#N.B. Only thing you need to change is line 7, below:
```{r}
individual_ID <- ""
```


```{r}
library("ggplot2") # for the plot
library("ggrepel") # for spreading text labels on the plot, you can replace with `geom_text` if you want
library("scales") # for axis labels notation
library("GWASTools")
library(karyoploteR) 
library(glue)
#https://stackoverflow.com/questions/33727432/how-to-plot-positions-along-a-chromosome-graphic
```

```{r}
rfmix <- read.table("RFMix_concat_chr_output.tsv",  header=T)
```

```{r}
hapA_column <- as.numeric(which(colnames(rfmix) == glue("{individual_ID}.0"))) #FIND WHERE INDIVIDUAL IS IN TABLE
hapB_column <- as.numeric(which(colnames(rfmix) == glue("{individual_ID}.1")))
individual <- rfmix[,c(1:6,{hapA_column},{hapB_column})]
names(individual)[1] <- "chromosome"
names(individual)[7] <- "hap_A"
names(individual)[8] <- "hap_B"

individual$pop1 <- ifelse(individual$hap_A == 0, "AFR",
                   ifelse(individual$hap_A == 1, "AMR",
                   ifelse(individual$hap_A == 2, "CAS",
                   ifelse(individual$hap_A == 3, "EAS",
                   ifelse(individual$hap_A == 4, "EUR",
                   ifelse(individual$hap_A == 5, "MDE",
                   ifelse(individual$hap_A == 6, "OCE",
                   ifelse(individual$hap_A == 7, "SAS", 'other'))))))))

individual$pop2 <- ifelse(individual$hap_B == 0, "AFR",
                   ifelse(individual$hap_B == 1, "AMR",
                   ifelse(individual$hap_B == 2, "CAS",
                   ifelse(individual$hap_B == 3, "EAS",
                   ifelse(individual$hap_B == 4, "EUR",
                   ifelse(individual$hap_B == 5, "MDE",
                   ifelse(individual$hap_B == 6, "OCE",
                   ifelse(individual$hap_B == 7, "SAS", 'other'))))))))
individual_hapA <- individual[,c(1:7,9)]
names(individual_hapA)[8] <- "Ancestry of Haplotype Block"
individual_hapB <- individual[,c(1:6,8,10)]
names(individual_hapB)[8] <- "Ancestry of Haplotype Block"
#rfmix.m <- colMeans(rfmix[, 2:ncol(rfmix)])

#tmp <- data.frame(X=rfmix.m, ind=rep(1:3))
#rfmix.df <- unstack(tmp, X~ind)

```

```{r}
#sizes from: https://www.ncbi.nlm.nih.gov/genome/gdv/browser/genome/?id=GCF_000001405.40
chrom_sizes <- structure(list(chromosome = c("chr1", "chr2", "chr3", "chr4", 
"chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11", "chr12", 
"chr13", "chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21", "chr22"), size = c(248956422L, 242193529L, 198295559L, 190214555L, 181538259L, 170805979L, 159345973L, 145138636L, 138394717L, 133797422L, 135086622L, 133275309L, 114364328L, 107043718L, 101991189L, 90338345L, 83257441L, 80373285L, 58617616L, 64444167L, 46709983L, 50818468L)), .Names = c("chromosome", "size"), class = "data.frame", row.names = c(NA, -22L))

#centromeres.hg38$left.base
centromeres <- structure(list(chromosome = c("chr1", "chr2", "chr3", "chr4", 
"chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11", "chr12", "chr13", "chr14", "chr15", "chr16", "chr17","chr18", "chr19", "chr20", "chr21", "chr22"), start = c(122026460L, 92188146L, 90772459L, 49708101L, 46485901L, 58553889L, 58169654L, 44033745L, 43236168L, 39686683L, 51078349L, 34769408L, 16000001L, 16000001L, 17000001L, 36311159L, 22813680L, 15460900L, 24498981L, 26436233L, 10864561L, 12954789L), end = c(125184587L, 94090557L, 93655574L, 51743951L, 50059807L, 59829934L, 60828234L, 45877265L, 45518558L, 41593521L, 54425074L, 37185252L, 18051248L, 18173523L, 19725254L, 38280682L, 26885980L, 20861206, 27190874L, 30038348L, 12915808L, 15054318L)), .Names = c("chromosome", 
"start", "end"), class = "data.frame", row.names = c(NA, -22L))

# create an ordered factor level to use for the chromosomes in all the datasets
chrom_order <- c("chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7","chr8", "chr9", "chr10", "chr11", "chr12", "chr13", "chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21", "chr22")

chrom_key <- setNames(object = as.character(c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 
                                              12, 13, 14, 15, 16, 17, 18, 19, 20, 
                                              21, 22)), nm = chrom_order)

# convert the chromosome column in each dataset to the ordered factor
chrom_sizes[["chromosome"]] <- factor(x = chrom_sizes[["chromosome"]], 
                                      levels = chrom_order)
#ancestry[["chromosome"]] <- factor(x = one[["chromosome"]], 
#                                     levels = chrom_order)
centromeres[["chromosome"]] <- factor(x = centromeres[["chromosome"]], 
                                      levels = chrom_order)

# create a color key for the plot
#group.colors <- c("0" = "red", "1" = "blue", "2" = "orange", "3" = "purple", "4" = "yellow", "5" =  "pink", "6" = "green", "7" = "black")
group.colors <- c("AFR" = "red", "AMR" = "blue", "CAS" = "orange", "EAS" = "purple", "EUR" = "yellow", "MDE" =  "pink", "OCE" = "green", "SAS" = "black")
```

```{r}
ggplot(data = chrom_sizes) + 
    # base rectangles for the chroms, with numeric value for each chrom on the x-axis
    geom_rect(aes(xmin = as.numeric(chromosome) + 0.1, 
                  xmax = as.numeric(chromosome) + 0.3, 
                  ymax = size, ymin = 0), 
              colour="black", fill = "white") + 
  
      geom_rect(aes(xmin = as.numeric(chromosome) - 0.1, 
                  xmax = as.numeric(chromosome) - 0.3, 
                  ymax = size, ymin = 0), 
              colour="black", fill = "white") + 
    # rotate the plot 90 degrees
    coord_flip() +
    # black & white color theme 
    theme(axis.text.x = element_text(colour = "black"), 
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(), 
          panel.background = element_blank()) + 
    # give the appearance of a discrete axis with chrom labels
    scale_x_discrete(name = "chromosome", limits = names(chrom_key)) +
    # add bands for centromeres
    geom_rect(data = centromeres, aes(xmin = as.numeric(chromosome) + 0.1, 
                                      xmax = as.numeric(chromosome) + 0.3, 
                                      ymax = end, ymin = start)) +
    geom_rect(data = centromeres, aes(xmin = as.numeric(chromosome) - 0.1, 
                                      xmax = as.numeric(chromosome) - 0.3, 
                                      ymax = end, ymin = start)) +
      # add bands for ancestry
    geom_rect(data = individual_hapA, aes(xmin = as.numeric(chromosome) + 0.1, 
                                     xmax = as.numeric(chromosome) + 0.3, 
                                     ymax = epos, ymin = spos, fill = `Ancestry of Haplotype Block`)) + 
  
    geom_rect(data = individual_hapB, aes(xmin = as.numeric(chromosome) - 0.1, 
                                     xmax = as.numeric(chromosome) - 0.3, 
                                     ymax = epos, ymin = spos, fill = `Ancestry of Haplotype Block`)) + 
  
    scale_fill_manual(values = group.colors) +

  #MAKE SURE TO CHANGE NAME OF INDIVIDUAL
    ggtitle(glue("Local Ancestry for ORIEN Sample {individual_ID}")) +
    # supress scientific notation on the y-axis
    scale_y_continuous(labels = comma) +
    ylab("region (bp)")

ggsave(filename=paste("Local_Ancestry_",{individual_ID},".jpeg",sep=""), width=10, height=10, units="in", res=1000, device = jpeg) 
```
